app/controllers/concerns/new_device_concern.rb

```diff
@@ -9,7 +9,10 @@ def set_new_device_session(new_device)
     user_session[:new_device] = new_device
   end
 
+  # @return [Boolean,nil] Whether current user session is from a new device. Returns nil if there is
```

Comment:
I feel like nil vs boolean is always a little tricky, did we consider like a symbol or string `:unknown` for the 3rd case?

Reply:
Hadn't considered it, no. I agree on the general point of tri-value types, but in terms of potential impact on usage of this value in `if new_device?` and logging, I think the `nil` value sorta works? i.e. not considered a new device, not logged one way or the other.

The other option is to revert back to something more like the original proposal in #10957, but use the `warden.authenticated?(:user)` logic in place of [the `success`-based ternary](https://github.com/18F/identity-idp/pull/10957/files#diff-513a92c45a3a5c14210895bd4082fb0d9b6a33dea5209ff25362f03d2a1f300bR179). That way, the return value of this method is only ever a boolean.

Reply:
f37b254 returns to a boolean-only return value.